libraries/mbed/common/rtc_time.c

```diff
@@ -58,6 +75,15 @@ clock_t clock() {
     return t;
 }
 
+void attach_rtc(time_t (*read_rtc)(void), void (*write_rtc)(time_t), void (*init_rtc)(void), int (*isenabled_rtc)(void)) {
+    _rtc_read = read_rtc;
+    _rtc_write = write_rtc;
+    _rtc_init = init_rtc;
+    _rtc_isenabled = isenabled_rtc;
+}
```

Comment:
Outside of init on which Martin already commented: Yes ideally it is atomic, and also the other RTC functions you don't want interrupted by attach. I might even do something nice with shadow variables. However: Is it really needed? There are a ton of mbed functions which can break when interrupting them (including for example changing pins of the same port to input/output). It seems to me to be in any normal use case a highly unlikely situation where this would interrupt anything: Normally you will call it at the start of your code.

For example usage, I published my test program (ignore lines 26/27, was debugging, not my own library I started with and helloworld was broken). This is using it with a C++ class, where for the functions I needed static member functions. If you do it purely C it would be easier, but this works fine. Should at least clarify how the attach_rtc is used: http://developer.mbed.org/users/Sissors/code/Adafruit_RTCLib/rev/09c202144d63. I don't think the init function will be used often in practise, easier to do that in the constructor.